overlord/devicestate/handlers_install.go

```diff
@@ -43,6 +43,28 @@ var (
 	sysconfigConfigureRunSystem = sysconfig.ConfigureRunSystem
 )
 
+func setSysconfigCloudOptions(opts *sysconfig.Options, gadgetDir string, model *asserts.Model) {
+	// 1. check cloud.cfg.d in the gadget snap, this is always ok regardless
+	//    of grade
+	cloudCfg := filepath.Join(gadgetDir, "cloud.cfg.d")
+	if osutil.IsDirectory(cloudCfg) {
+		opts.CloudInitSrcDir = cloudCfg
```

Comment:
What if both locations are present?

Reply:
A good question, right now the gadget wins but I guess that is a matter for debate and we could allow more customization for dangerous models.

Reply:
Maybe in a followup so that cloud people can prepare images now?

Reply:
Yeah, a follow up is fine. Let's unblock the cloud guys first.

Reply:
Hm maybe we should leave a TODO at least.